Mathlib/LinearAlgebra/Multilinear/TensorProduct.lean

```diff
@@ -22,6 +22,47 @@ variable {R ι₁ ι₂ ι₃ ι₄ : Type*}
 variable [CommSemiring R]
 variable {N₁ : Type*} [AddCommMonoid N₁] [Module R N₁]
 variable {N₂ : Type*} [AddCommMonoid N₂] [Module R N₂]
+
+attribute [local simp] add_tmul tmul_add smul_tmul
+
+section
+
+variable {N : ι₁ ⊕ ι₂ → Type*} [∀ i, AddCommMonoid (N i)] [∀ i, Module R (N i)]
+
+/-- Given a family of modules `N` indexed by a type `ι₁ ⊕ ι₂`,
+a multilinear map from the modules `N (.inl i₁)` to `N₁` and
+a multilinear map from the modules `N (.inr i₁)` to `N₂`, this
+is the induced multilinear map from all the modules `N i` to `N₁ ⊗ N₂`. -/
+@[simps apply]
+def domCoprodDep (a : MultilinearMap R (fun i₁ ↦ N (.inl i₁)) N₁)
+    (b : MultilinearMap R (fun i₂ ↦ N (.inr i₂)) N₂) :
+    MultilinearMap R N (N₁ ⊗[R] N₂) where
+  toFun v := a (fun i₁ ↦ v (.inl i₁)) ⊗ₜ b (fun i₂ ↦ v (.inr i₂))
```

Comment:
Out of scope for this PR, but: I think I made the wrong choice long ago here, and we should have used
```suggestion
  toFun v := f (a (fun i₁ ↦ v (.inl i₁))) (b (fun i₂ ↦ v (.inr i₂)))
```
rather than picking the tensor product as the bilinear map.

(Obviously the tensor product is universal, but it's in practice easier to save the caller going through it, and this also reduces the dependencies of this definition significantly).